### DISCUSSION TITLE
Bluetooh headset connect/disconnect events reset keyboard layout

### DISCUSSION THREAD

**Discussion by pvonmoradi** [blueman-project/blueman#2501] [Q&A]
blueman: 2.3.5-2
BlueZ: 5.66-1
Distribution: Debian 12
Desktop environment: i3wm

I set my keyboard to have two layouts using this command: `setxkbmap -layout us,ir -option ctrl:nocaps;` in a startup script. It works fine, but, when a bluetooth headset device is connected or disconnected, my keyboard layout resets to us only! It's strange cause I don't see how a bluetooth headset is related to keyboard input layouts. 
(To be honest, I'm not sure whether this problem is related to blueman, I wonder if this applet somehow runs some post-connect script that may reset X input layout)

**Comment by infirit** [comment]
Blueman doesn't do anything like changing layouts. What I can imagine is that the headset is exposing some sort of input device (buttons?) which triggers other parts of the system on (dis)connect to re-evaluate the layout. Converting to a discussion.

**Comment by jordr** [comment]
I can confirm, this happens anytime you connect a new device like screens etc. I don't think tthis is on Blueman.
